src/components/SingleArticle/SingleArticle.jsx

```diff
@@ -0,0 +1,79 @@
+import Header from "../Header";
+import { useParams } from "react-router-dom";
+import { viewSingleArticle, getComments } from "../../utils/api";
+import { useState, useEffect } from "react";
+import "./SingleArticle.css";
+
+export default function SingleArticle() {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState([]);
+  const [comments, setComments] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const articleDate = new Date(article.created_at).toString();
+
+  useEffect(() => {
+    viewSingleArticle(article_id)
+      .then((response) => {
+        setArticle(response);
+      })
+      .catch((err) => {
+        setError(err);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+      
+    getComments(article_id)
+      .then((response) => {
+        setComments(response);
+      })
+      .catch((err) => {
+        setError(err);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [article_id]);
+
+  if (loading) {
+    return <p>Loading...</p>;
+  }
+
+  if (error) {
+    return <p>Error: {error.message}</p>;
+  }
+
+  return (
+    <>
+      <Header />
```

Comment:
Is it worth moving the header into the App.jsx file and putting it above the routes ? That way it'll render on every view for you, so that you do not need to paste this here every time in all components.